exporter/opentelemetry-exporter-richconsole/src/opentelemetry/exporter/richconsole/__init__.py

```diff
@@ -173,14 +173,15 @@ def spans_to_tree(spans: typing.Sequence[ReadableSpan]) -> Dict[str, Tree]:
         trees = {}
         parents = {}
         spans = list(spans)
+        span_ids = {s.context.span_id for s in spans}
         while spans:
             for span in spans:
-                if not span.parent:
+                if not span.parent or span.parent.span_id not in span_ids:
```

Comment:
Make sure that the parent is part of the spans. If not, treat the child span as root span.